src/main/java/com/blackducksoftware/integration/hub/alert/channel/hipchat/HipChatChannel.java

```diff
@@ -73,6 +81,41 @@ public String getApiUrl() {
         return HIP_CHAT_API;
     }
 
+    @Override
+    public String testGlobalConfig(final GlobalHipChatConfigEntity entity) {
+        if (entity == null) {
+            return "The provided entity was null.";
+        }
+        final RestConnection restConnection = channelRestConnectionFactory.createUnauthenticatedRestConnection(HIP_CHAT_API);
+        if (restConnection != null) {
+            final List<String> urlSegments = Arrays.asList("v2", "room", "*", "notification");
+            final Map<String, String> queryParameters = new HashMap<>();
+            queryParameters.put("auth_test", "true");
+
+            final Map<String, String> requestHeaders = new HashMap<>();
+            requestHeaders.put("Authorization", "Bearer " + entity.getApiKey());
+            requestHeaders.put("Content-Type", "application/json");
+
+            final HttpUrl httpUrl = restConnection.createHttpUrl(urlSegments, queryParameters);
+            // The body = {"message":"test"} is required to avoid a BAD_REQUEST (OkHttp issue: #854)
+            final RequestBody body = restConnection.createJsonRequestBody("{\"message\":\"test\"}");
```

Comment:
Can you use the createMessageRequest from ChannelRequestHelper to help encapsulate creating the request body and the post request.

Reply:
Yes. My mistake was adding this before I added that method in ChannelRequestHelper.